electrum/submarine_swaps.py

```diff
@@ -171,6 +173,30 @@ def start_network(self, *, network: 'Network', lnwatcher: 'LNWalletWatcher'):
                 continue
             self.add_lnwatcher_callback(swap)
 
+    async def pay_pending_invoices(self):
+        # for server
+        self.invoices_to_pay = set()
+        while True:
+            await asyncio.sleep(1)
+            for key in list(self.invoices_to_pay):
+                swap = self.swaps.get(key)
+                if not swap:
+                    continue
+                invoice = self.wallet.get_invoice(key)
+                if not invoice:
+                    continue
+                current_height = self.network.get_local_height()
+                delta = swap.locktime - current_height
+                if delta <= MIN_LOCKTIME_DELTA:
+                    # fixme: should consider cltv of ln payment
+                    self.logger.info(f'locktime too close {key}')
+                    continue
+                success, log = await self.lnworker.pay_invoice(invoice.lightning_invoice, attempts=1)
+                if not success:
+                    self.logger.info(f'failed to pay invoice {key}')
+                    continue
+                self.invoices_to_pay.remove(key)
+
     @log_exceptions
     async def _claim_swap(self, swap: SwapData) -> None:
```

Comment:
This func has many conditional code blocks on `swap.is_reverse`. For readability and maintainability it might be better to split code for forward swaps and reverse swaps.

Reply:
fair enough. I will restructure that